flink-libraries/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/CliResultView.java

```diff
@@ -150,9 +151,10 @@ protected void openRow() {
 		view.open(); // enter view
 	}
 
-	protected void stopRetrieval() {
+	protected void stopRetrieval(TypedResult.ResultType jobStatus) {
 		// stop retrieval
 		refreshThread.isRunning = false;
+		refreshThread.jobStatus = jobStatus;
```

Comment:
Switch these two statements because otherwise the refresh thread can already do side effects when `isRunning` is set to false.